cclient/response/get_tips.c

```diff
@@ -7,14 +7,39 @@
 
 #include "response/get_tips.h"
 
-get_tips_res_t* get_tips_res_new() { return flex_hash_array_new(); }
+get_tips_res_t* get_tips_res_new() {
+  get_tips_res_t* res = (get_tips_res_t*)malloc(sizeof(get_tips_res_t));
+  if (res) {
+    res->tips = flex_hash_array_new();
+  }
+  return res;
+}
+
+void get_tips_res_free(get_tips_res_t** const res) {
+  if (!res || !(*res)) {
+    return;
+  }
+
+  get_tips_res_t* tmp = *res;
 
-trit_array_p get_tips_res_hash_at(get_tips_res_t* hashes, size_t index) {
-  return flex_hash_array_at(hashes, index);
+  if (tmp->tips) {
+    flex_hash_array_free(tmp->tips);
+  }
+  free(tmp);
+  *res = NULL;
 }
 
-int get_tips_res_hash_num(get_tips_res_t* hashes) {
-  return flex_hash_array_count(hashes);
+get_tips_res_t* get_tips_res_add_tip(get_tips_res_t* const res,
+                                     tryte_t const* const tip) {
```

Comment:
We should take `flex_trit` here and only serialize to string when converting to JSON

Reply:
e.g. see below where we erroneously convert tryte_t to trit_array_p all of a sudden

Reply:
@oopsmonk  is doing it in #473